frontend/src/components/widgets/TimeInput/TimeInput.tsx

```diff
@@ -108,8 +108,11 @@ class TimeInput extends PureComponent<Props, State> {
     )
   }
 
-  private handleChange = (newDate: Date): void => {
-    const value = this.dateToString(newDate)
+  private handleChange = (newDate: Date | null): void => {
+    let value = this.initialValue
```

Comment:
I would propose a small refactoring here:
```typescript
    let value: string
    if (newDate === null) {
      // This case is not supposed to happen since time picker cannot be cleared.
      value = this.initialValue
    } else {
      value = this.dateToString(newDate)
    }
```
The reason is that `handleChange` could be called lots of times and `initialValue` is doing some still with widget manager which might be a bit more expensive (not sure). Also added a comment since this scenario should actually never happen with the current implementation.

Reply:
Good call out 🥇 It's changed 👍